app/main.py

```diff
@@ -1,3 +1,15 @@
-def cache(func):
-    # Write your code here
-    pass
+from typing import Callable
+
+
+def cache(func: Callable) -> Callable:
+    result_args_dict = {}
+
+    def inner(*args) -> Callable:
```

Comment:
You could consider using decorator from `functools` for `inner` to preserve the name and docstring of a function this decorator will be applied to.